infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -182,23 +182,18 @@ public ServerContext(Map<String, Object> serverConfig) {
     }
 
     int getBaseServerThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--base-server-threads");
-        return threadCount == null ? 1 : threadCount;
-    }
-
-    int getLayoutServerThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--layout-server-threads");
-        return threadCount == null ? 1 : threadCount;
+        int threadCount = Integer.parseInt(getServerConfig(String.class, "--base-server-threads"));
+        return threadCount == 0 ? 1 : threadCount;
     }
 
     public int getLogunitThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--logunit-threads");
-        return threadCount == null ? Runtime.getRuntime().availableProcessors() * 2 : threadCount;
+        int threadCount = Integer.parseInt(getServerConfig(String.class, "--logunit-threads"));
+        return threadCount == 0 ? Runtime.getRuntime().availableProcessors() * 2 : threadCount;
     }
 
     public int getManagementServerThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--management-server-threads");
-        return threadCount == null ? 4 : threadCount;
+        Integer threadCount = Integer.parseInt(getServerConfig(String.class, "--management-server-threads"));
+        return threadCount == 0 ? 4 : threadCount;
```

Comment:
Do we want to unbox here as well?

Reply:
The initial was interpreting null as 0, so I just changed it to a native int instead for simplicity.

Reply:
Yeah I see, I mean the type in line 195 could also be int instead of Integer I thought?

Reply:
Yea, you are right. Its not needed, removed.